scripts/variantstore/wdl/extract/hail_create_vat_inputs.py

```diff
@@ -122,39 +122,55 @@ def write_sites_only_vcf(ac_an_af_split, sites_only_vcf_path):
     t = mt.rows()
-    t.select(var_origin_id=hl.format('%s-%s-%s-%s', t.locus.contig, t.locus.position, t.alleles[0], t.alleles[1])).export(var_ids_path, parallel='header_per_shard')
+    t.select(var_origin_id=hl.format('%s-%s-%s-%s', t.locus.contig, t.locus.position, t.alleles[0], t.alleles[1])).export(var_ids_path, parallel='separate_header')
 
 def main(vds, ancestry_file_location, sites_only_vcf_path):
-    transforms = [
-        hard_filter_non_passing_sites,
-        failing_gts_to_no_call,
-        remove_too_many_alt_allele_sites
-    ]
-    transformed_vds=vds
-    for transform in transforms:
-        transformed_vds = transform(transformed_vds)
+    n_parts = vds.variant_data.n_partitions()
+    n_rounds = 5
+    parts_per_round = n_parts // n_rounds
+    ht_paths = [sites_only_vcf_path.replace(r".sites-only.vcf.bgz", f'_{i}.ht') for i in range(n_rounds)]
+    for i in range(n_rounds):
+        part_range = range(i*parts_per_round, min((i+1)*parts_per_round, n_parts))
+        vds_part = hl.vds.VariantDataset(
+            vds.reference_data._filter_partitions(part_range),
+            vds.variant_data._filter_partitions(part_range),
+        )
+
+        transforms = [
+            hard_filter_non_passing_sites,
+            failing_gts_to_no_call,
+            remove_too_many_alt_allele_sites
+        ]
+        transformed_vds=vds_part
+        for transform in transforms:
+            transformed_vds = transform(transformed_vds)
+
+        mt = hl.vds.to_dense_mt(transformed_vds)
 
-    mt = hl.vds.to_dense_mt(transformed_vds)
+        with open(ancestry_file_location, 'r') as ancestry_file:
+            mt = matrix_table_ac_an_af(mt, ancestry_file) # this adds subpopulation information and splits our multi-allelic rows
 
-    with open(ancestry_file_location, 'r') as ancestry_file:
-        mt = matrix_table_ac_an_af(mt, ancestry_file) # this adds subpopulation information and splits our multi-allelic rows
+        ht = mt.rows()
+        ht = ht.select('call_stats_by_pop', 'a_index', 'ac_an_af')
+        ht.write(ht_paths[i])
 
-    # potentially in the future: merge AC, AN, AF back to the original VDS with: vds = vds_ac_an_af(mt, vds)
+        # potentially in the future: merge AC, AN, AF back to the original VDS with: vds = vds_ac_an_af(mt, vds)
 
-    # for debugging information
-    add_variant_tracking_info(mt, sites_only_vcf_path)
+        # for debugging information -- remove for now to get us through Echo
+        # add_variant_tracking_info(mt, sites_only_vcf_path)
 
     # create a sites only VCF (that is hard filtered!) and that can be made into a custom annotations TSV for Nirvana to use with AC, AN, AF, SC for all subpopulations and populations
-    write_sites_only_vcf(mt, sites_only_vcf_path)
+    ht_list = [hl.read_table(ht_path) for ht_path in ht_paths]
+    ht_all = ht_list[0].union(*ht_list[1:])
+    write_sites_only_vcf(ht_all, sites_only_vcf_path)
```

Comment:
this is magical to me... can we add some comments to explain what's happening here?